tests/api/test_user_endpoints.py

```diff
@@ -0,0 +1,721 @@
+@pytest.mark.parametrize("auth_header", [[STORAGE_READ]], indirect=True)
+@pytest.mark.integration
+def test_create_user_wrong_scope(
+    test_client: TestClient, auth_header: dict[str, str]
+) -> None:
+    response = test_client.post(URL, headers=auth_header, json={})
+    assert HTTP_403_FORBIDDEN == response.status_code
+
+
+@pytest.mark.usefixtures("db")
+@pytest.mark.parametrize("auth_header", [[USER_CREATE]], indirect=True)
+@pytest.mark.integration
+def test_create_user_bad_username(
+    test_client: TestClient, auth_header: dict[str, str]
+) -> None:
+    body = {
+        "username": "spaces in name",
+        "password": str_to_b64_str("TestP@ssword9"),
+    }
+
+    response = test_client.post(URL, headers=auth_header, json=body)
+    assert HTTP_422_UNPROCESSABLE_ENTITY == response.status_code
+
+
+@pytest.mark.parametrize(
+    "password, expected",
+    [
+        ("short", "at least eight characters"),
+        ("longerpassword", "at least one number"),
+        ("longer55pasword", "at least one capital letter"),
+        ("LoNgEr55paSSword", "at least one symbol"),
+    ],
+)
+@pytest.mark.usefixtures("db")
+@pytest.mark.parametrize("auth_header", [[USER_CREATE]], indirect=True)
+@pytest.mark.integration
+def test_create_user_bad_password(
+    password: str, expected: str, test_client: TestClient, auth_header: dict[str, str]
+) -> None:
+    body = {"username": "test_user", "password": str_to_b64_str(password)}
+    response = test_client.post(URL, headers=auth_header, json=body)
+    assert HTTP_422_UNPROCESSABLE_ENTITY == response.status_code
+    assert expected in response.json()["detail"][0]["msg"]
+
+
+@pytest.mark.parametrize("auth_header", [[USER_CREATE]], indirect=True)
+@pytest.mark.integration
+def test_username_exists(
+    test_client: TestClient, auth_header: dict[str, str], db: Session
+) -> None:
+    body = {"username": "test_user", "password": str_to_b64_str("TestP@ssword9")}
+    user = FidesUser.create(db=db, data=body)
+
+    response = test_client.post(URL, headers=auth_header, json=body)
+    response_body = json.loads(response.text)
+    assert response_body["detail"] == "Username already exists."
+    assert HTTP_400_BAD_REQUEST == response.status_code
+
+    user.delete(db)
+
+
+@pytest.mark.integration
+def test_delete_user_not_authenticated(
+    test_client: TestClient, user: FidesUser
+) -> None:
+    url = f"{URL}/{user.id}"
+    response = test_client.delete(url, headers={})
+    assert HTTP_401_UNAUTHORIZED == response.status_code
+
+
+@pytest.mark.usefixtures("db")
+@pytest.mark.parametrize("auth_header", [[USER_DELETE]], indirect=True)
+@pytest.mark.integration
+def test_delete_user_not_admin_root_or_self(
+    test_client: TestClient, auth_header: dict[str, str], user: FidesUser
+) -> None:
+    url = f"{URL}/{user.id}"
+    response = test_client.delete(url, headers=auth_header)
+    assert HTTP_403_FORBIDDEN == response.status_code
+
+
+@pytest.mark.usefixtures("db")
```

Comment:
why `usefixtures` here instead of calling it directly in the function declaration?

Reply:
It prevents linters from complaining about unused parameters. So `usefixtures` when the fixture is needed but not directly accessed. In the function declaration when it is being accessed.